### PR TITLE
Dialog Editor - reset default_value when changing dropdown type, values, or between single and multiselect

### PR DESCRIPTION
**Pull request by mzazrivec**
1. Automation ➻ Automate ➻ Customization
2. On "All Dialogs" ➻ Button Configuration ➻ Add new dialog 
3. Add a dropdownlist to the "new section"
4. Edit the dropdownlist
5. Leave the default 1,2,3 options
6. Change value type to INTEGER, set the default to "One"
7. Also set REQUIERED to Yes.
8. Save and then Save the dialog
9. Select dialog test5 and Configuration ➻ edit this dialog
10. Edit the dropdownlist
11. In Options, change MULTISELECT to Yes
12. Save and then Save the dialog
13. Observe the following error:

Screenshot from 2020-04-21 14-47-42

Closes https://github.com/ManageIQ/ui-components/pull/432

https://bugzilla.redhat.com/show_bug.cgi?id=1746211

The PR also updates node.js in travis to version 10 to fix travis failures.

### CODE REVIEW

File: src/dialog-editor/components/modal-field-template/modalFieldTemplateComponent.ts [hunk 1]
```diff
@@ -28,14 +28,46 @@ class ModalFieldController {
...
+        return field.options.force_multi_value ? [] : byDataType;
+      default:
+        return byDataType;
     }
   }
 
+  public resetDefaultValue() {
+    // TODO first use the real value if possible
+    this.modalData.default_value = this.emptyDefaultValue(this.modalData);
+    console.log('resetDefaultValue', this.modalData.default_value);
```
Comment (himdel) [inline src/dialog-editor/components/modal-field-template/modalFieldTemplateComponent.ts:48]: console.log

**Comment by himdel** [comment]
Works for me, thanks :)
Just that one extra console.log..

**Comment by mzazrivec** [comment]
> Just that one extra console.log..

Removed, thank you.

**Comment by simaishi** [comment]
Ivanchuk backport details:

```
$ git log -1
commit 541f3e4eb5f0bd99fd920b3f4bee22c2deffb184
Author: Martin Hradil <mhradil@redhat.com>
Date:   Wed Apr 22 18:00:52 2020 +0200

    Merge pull request #437 from mzazrivec/bug-1746211

    Dialog Editor - reset default_value when changing dropdown type, values, or between single and multiselect

    (cherry picked from commit 2a03d66e50dabb9e9d04185fdb2b7881c1816237)

    https://bugzilla.redhat.com/show_bug.cgi?id=1746211
```

**Comment by simaishi** [comment]
Jansa backport details:

```
$ git log -1
commit a6c6ebbef2449048ad077987bb3af86570945f3c
Author: Martin Hradil <mhradil@redhat.com>
Date:   Wed Apr 22 18:00:52 2020 +0200

    Merge pull request #437 from mzazrivec/bug-1746211

    Dialog Editor - reset default_value when changing dropdown type, values, or between single and multiselect

    (cherry picked from commit 2a03d66e50dabb9e9d04185fdb2b7881c1816237)

    https://bugzilla.redhat.com/show_bug.cgi?id=1746211
```

**Pull request merged**